app/config.go

```diff
@@ -222,3 +243,87 @@ func (s *ServerToml) String() string {
 	}
 	return buff.String()
 }
+
+// CertificateURLType represents the type of a CertificateURL.
+// The supported types are defined as constants of type CertificateURLType.
+type CertificateURLType string
+
+// CertificateURL contains the CertificateURLType and the actual URL
+// certificate, which can be a path leading to a file containing a certificate
+// or a string directly being a certificate.
+type CertificateURL string
+
+const (
+	// String is a CertificateURL type containing a certificate.
+	String CertificateURLType = "string"
```

Comment:
Another proposal: the type should default to file, so that WebSocketTLSCertificate = "/etc/conode.crt" will do what any reasonable person would expect.

But since it is already here, WebSocketTLSCertificate = "string:XXX" should work too. (Although it is unclear that YAML can accept the embedded newlines that you typically find in PEM-encoded cert files.)

I do like the type:blob format, because it will help with LetsEncrypt. We can add LetsEncrypt support here like this: WebSocketTLSCertificate = "le:hostname.example.com" (LE requires that the server know what it's public FQDN is at the moment it requests the certificate.)

Reply:
I see! Yes I can do that. Also, are you saying we should change the delimiter from "://" to ":", or is "://" still fine?

Reply:
No, that's a mistake. I propose that the following work:
- "filename.pem"
- "file://filename.pem"
- "string://certificate-goes-here"
- "letsencrypt://FQDN-goes-here" (later, in another commit)